src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1658,6 +1660,9 @@ private static SyntaxTokenList AsModifierList(Accessibility accessibility, Decla
             if (modifiers.IsExtern)
                 list.Add(SyntaxFactory.Token(SyntaxKind.ExternKeyword));
 
+            if (modifiers.IsRequired)
```

Comment:
Do any .NET design guidelines need to be updated to recommend a placement of the 'required' modifier?